### ISSUE TITLE
After successful login, the interface is blank

### ISSUE THREAD

**Issue by HueLiu** [lujjjh/lito#2]
After successful login, the interface is blank.

lito version: v0.1.0.0-beta.3
Windows version: 21H2 Pro
Apple ID location: Hong Kong

```
GET https://app.example/components/musickit-components/locales/zh-cn/translations.json net::ERR_ABORTED 404 (Not Found)
loadTranslations @ p-6b1b37da.js:1
initialize @ p-6b1b37da.js:1
k @ p-6b1b37da.js:1
componentWillLoad @ p-21c59437.entry.js:1
r @ p-1ef6431f.js:1
e.componentWillLoad @ p-1ef6431f.js:1
async function(异步)
e.componentWillLoad @ p-1ef6431f.js:1
ne @ p-a23b3d3a.js:461
K @ p-a23b3d3a.js:342
(匿名) @ p-a23b3d3a.js:334
ke @ p-a23b3d3a.js:707
ve @ p-a23b3d3a.js:714
requestAnimationFrame(异步)
raf @ p-a23b3d3a.js:17
(匿名) @ p-a23b3d3a.js:702
J @ p-a23b3d3a.js:334
i @ p-a23b3d3a.js:638
(匿名) @ p-a23b3d3a.js:639
async function(异步)
(匿名) @ p-a23b3d3a.js:611
(匿名) @ p-a23b3d3a.js:640
(匿名) @ p-a23b3d3a.js:644
jmp @ p-a23b3d3a.js:16
connectedCallback @ p-a23b3d3a.js:577
(匿名) @ p-a23b3d3a.js:669
re @ p-a23b3d3a.js:669
(匿名) @ musickit-components.esm.js:12
Promise.then(异步)
(匿名) @ musickit-components.esm.js:11
p-6b1b37da.js:1 GET https://app.example/components/musickit-components/locales/en/translations.json net::ERR_ABORTED 404 (Not Found)
loadTranslations @ p-6b1b37da.js:1
async function(异步)
loadTranslations @ p-6b1b37da.js:1
initialize @ p-6b1b37da.js:1
k @ p-6b1b37da.js:1
componentWillLoad @ p-21c59437.entry.js:1
r @ p-1ef6431f.js:1
e.componentWillLoad @ p-1ef6431f.js:1
async function(异步)
e.componentWillLoad @ p-1ef6431f.js:1
ne @ p-a23b3d3a.js:461
K @ p-a23b3d3a.js:342
(匿名) @ p-a23b3d3a.js:334
ke @ p-a23b3d3a.js:707
ve @ p-a23b3d3a.js:714
requestAnimationFrame(异步)
raf @ p-a23b3d3a.js:17
(匿名) @ p-a23b3d3a.js:702
J @ p-a23b3d3a.js:334
i @ p-a23b3d3a.js:638
(匿名) @ p-a23b3d3a.js:639
async function(异步)
(匿名) @ p-a23b3d3a.js:611
(匿名) @ p-a23b3d3a.js:640
(匿名) @ p-a23b3d3a.js:644
jmp @ p-a23b3d3a.js:16
connectedCallback @ p-a23b3d3a.js:577
(匿名) @ p-a23b3d3a.js:669
re @ p-a23b3d3a.js:669
(匿名) @ musickit-components.esm.js:12
Promise.then(异步)
(匿名) @ musickit-components.esm.js:11
p-6b1b37da.js:1 GET https://app.example/components/musickit-components/locales/en-gb/translations.json net::ERR_ABORTED 404 (Not Found)
loadTranslations @ p-6b1b37da.js:1
async function(异步)
loadTranslations @ p-6b1b37da.js:1
initialize @ p-6b1b37da.js:1
k @ p-6b1b37da.js:1
componentWillLoad @ p-21c59437.entry.js:1
r @ p-1ef6431f.js:1
e.componentWillLoad @ p-1ef6431f.js:1
async function(异步)
e.componentWillLoad @ p-1ef6431f.js:1
ne @ p-a23b3d3a.js:461
K @ p-a23b3d3a.js:342
(匿名) @ p-a23b3d3a.js:334
ke @ p-a23b3d3a.js:707
ve @ p-a23b3d3a.js:714
requestAnimationFrame(异步)
raf @ p-a23b3d3a.js:17
(匿名) @ p-a23b3d3a.js:702
J @ p-a23b3d3a.js:334
i @ p-a23b3d3a.js:638
(匿名) @ p-a23b3d3a.js:639
async function(异步)
(匿名) @ p-a23b3d3a.js:611
(匿名) @ p-a23b3d3a.js:640
(匿名) @ p-a23b3d3a.js:644
jmp @ p-a23b3d3a.js:16
connectedCallback @ p-a23b3d3a.js:577
(匿名) @ p-a23b3d3a.js:669
re @ p-a23b3d3a.js:669
(匿名) @ musickit-components.esm.js:12
Promise.then(异步)
(匿名) @ musickit-components.esm.js:11
p-6b1b37da.js:1 GET https://app.example/components/musickit-components/locales/en-us/translations.json net::ERR_ABORTED 404 (Not Found)
loadTranslations @ p-6b1b37da.js:1
async function(异步)
loadTranslations @ p-6b1b37da.js:1
initialize @ p-6b1b37da.js:1
k @ p-6b1b37da.js:1
componentWillLoad @ p-21c59437.entry.js:1
r @ p-1ef6431f.js:1
e.componentWillLoad @ p-1ef6431f.js:1
async function(异步)
e.componentWillLoad @ p-1ef6431f.js:1
ne @ p-a23b3d3a.js:461
K @ p-a23b3d3a.js:342
(匿名) @ p-a23b3d3a.js:334
ke @ p-a23b3d3a.js:707
ve @ p-a23b3d3a.js:714
requestAnimationFrame(异步)
raf @ p-a23b3d3a.js:17
(匿名) @ p-a23b3d3a.js:702
J @ p-a23b3d3a.js:334
i @ p-a23b3d3a.js:638
(匿名) @ p-a23b3d3a.js:639
async function(异步)
(匿名) @ p-a23b3d3a.js:611
(匿名) @ p-a23b3d3a.js:640
(匿名) @ p-a23b3d3a.js:644
jmp @ p-a23b3d3a.js:16
connectedCallback @ p-a23b3d3a.js:577
(匿名) @ p-a23b3d3a.js:669
re @ p-a23b3d3a.js:669
(匿名) @ musickit-components.esm.js:12
Promise.then(异步)
(匿名) @ musickit-components.esm.js:11
6[Violation]Added non-passive event listener to a scroll-blocking <某些> 事件. Consider marking event handler as 'passive' to make the page more responsive. See <URL>
```

```
p-6b1b37da.js:1 GET https://app.example/components/musickit-components/locales/en-gb/translations.json 404 (Not Found)
loadTranslations @ p-6b1b37da.js:1
async function(异步)
loadTranslations @ p-6b1b37da.js:1
initialize @ p-6b1b37da.js:1
k @ p-6b1b37da.js:1
componentWillLoad @ p-21c59437.entry.js:1
r @ p-1ef6431f.js:1
e.componentWillLoad @ p-1ef6431f.js:1
async function(异步)
e.componentWillLoad @ p-1ef6431f.js:1
ne @ p-a23b3d3a.js:461
K @ p-a23b3d3a.js:342
(匿名) @ p-a23b3d3a.js:334
ke @ p-a23b3d3a.js:707
ve @ p-a23b3d3a.js:714
requestAnimationFrame(异步)
raf @ p-a23b3d3a.js:17
(匿名) @ p-a23b3d3a.js:702
J @ p-a23b3d3a.js:334
i @ p-a23b3d3a.js:638
(匿名) @ p-a23b3d3a.js:639
async function(异步)
(匿名) @ p-a23b3d3a.js:611
(匿名) @ p-a23b3d3a.js:640
(匿名) @ p-a23b3d3a.js:644
jmp @ p-a23b3d3a.js:16
connectedCallback @ p-a23b3d3a.js:577
(匿名) @ p-a23b3d3a.js:669
re @ p-a23b3d3a.js:669
(匿名) @ musickit-components.esm.js:12
Promise.then(异步)
(匿名) @ musickit-components.esm.js:11
2vendor.70de7908.js:27 TypeError: Cannot read properties of undefined (reading 'url')
    at L (index.5c77e547.js:75)
    at Hi (vendor.70de7908.js:27)
    at Nu (vendor.70de7908.js:27)
    at ps (vendor.70de7908.js:27)
    at fs (vendor.70de7908.js:27)
    at cs (vendor.70de7908.js:27)
    at ns (vendor.70de7908.js:27)
    at vendor.70de7908.js:27
    at e.unstable_runWithPriority (vendor.70de7908.js:18)
    at Ta (vendor.70de7908.js:27)
ql @ vendor.70de7908.js:27
Yl.n.callback @ vendor.70de7908.js:27
Ja @ vendor.70de7908.js:27
eu @ vendor.70de7908.js:27
vs @ vendor.70de7908.js:27
e.unstable_runWithPriority @ vendor.70de7908.js:18
Ta @ vendor.70de7908.js:27
gs @ vendor.70de7908.js:27
ns @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
e.unstable_runWithPriority @ vendor.70de7908.js:18
Ta @ vendor.70de7908.js:27
ja @ vendor.70de7908.js:27
Ra @ vendor.70de7908.js:27
Xu @ vendor.70de7908.js:27
hl @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
e.setState @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
Promise.then(异步)
e._run @ vendor.70de7908.js:27
e.run @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
ks @ vendor.70de7908.js:27
e.unstable_runWithPriority @ vendor.70de7908.js:18
Ta @ vendor.70de7908.js:27
ys @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
I @ vendor.70de7908.js:18
b.port1.onmessage @ vendor.70de7908.js:18
index.html#/:1 [Intervention]Images loaded lazily and replaced with placeholders. Load events are deferred. See https://go.microsoft.com/fwlink/?linkid=2048113
vendor.70de7908.js:27 TypeError: Cannot read properties of undefined (reading 'url')
    at L (index.5c77e547.js:75)
    at Hi (vendor.70de7908.js:27)
    at Nu (vendor.70de7908.js:27)
    at ps (vendor.70de7908.js:27)
    at fs (vendor.70de7908.js:27)
    at cs (vendor.70de7908.js:27)
    at ns (vendor.70de7908.js:27)
    at vendor.70de7908.js:27
    at e.unstable_runWithPriority (vendor.70de7908.js:18)
    at Ta (vendor.70de7908.js:27)
(匿名) @ vendor.70de7908.js:27
Promise.catch(异步)
e._run @ vendor.70de7908.js:27
e.run @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
ks @ vendor.70de7908.js:27
e.unstable_runWithPriority @ vendor.70de7908.js:18
Ta @ vendor.70de7908.js:27
ys @ vendor.70de7908.js:27
(匿名) @ vendor.70de7908.js:27
I @ vendor.70de7908.js:18
b.port1.onmessage @ vendor.70de7908.js:18
index.html#/:1 Uncaught (in promise) useRequest has caught the exception, if you need to handle the exception yourself, you can set options.throwOnError to true.
```
![image](https://user-images.githubusercontent.com/19755727/134481583-f9bfca55-5acb-4a98-ac83-5e1426deb773.png)

![image](https://user-images.githubusercontent.com/19755727/134480355-2dec01f9-e079-439e-bdb0-e716482106d7.png)

**Comment by lujjjh** [comment]
感谢吃螃蟹 :)

I believe there is something in the recommendations that I doesn't handle correctly...

Could you please evaluate

```js
[].concat.apply([], (await MusicKit.getInstance().api.music('v1/me/recommendations?timezone=%2B08:00&with=friendsMix%2Clibrary&name=listen-now&art%5Burl%5D=c%2Cf&omit%5Bresource%5D=autos&relate%5Beditorial-items%5D=contents&extend=editorialCard%2CeditorialVideo&extend%5Balbums%5D=artistUrl&extend%5Bplaylists%5D=artistNames%2CeditorialArtwork&extend%5Bsocial-profiles%5D=topGenreNames&include%5Balbums%5D=artists&include%5Bsongs%5D=artists&include%5Bmusic-videos%5D=artists&fields%5Balbums%5D=artistName%2CartistUrl%2Cartwork%2CcontentRating%2CeditorialArtwork%2CeditorialVideo%2Cname%2CplayParams%2CreleaseDate%2Curl&fields%5Bartists%5D=name%2Curl&extend%5Bstations%5D=airDate%2CsupportsAirTimeUpdates&meta%5Bstations%5D=inflectionPoints&types=artists%2Calbums%2Ceditorial-items%2Clibrary-albums%2Clibrary-playlists%2Cmusic-movies%2Cmusic-videos%2Cplaylists%2Cstations%2Cuploaded-audios%2Cuploaded-videos%2Cactivities%2Capple-curators%2Ccurators%2Ctv-shows&relate=catalog&platform=web')).data.data.map(recommendation => recommendation?.relationships?.contents?.data?.filter(item => !item?.attributes?.artwork)))
```

in the console and paste the result?

This will fillter out the recommendation items without artworks and I'd like to see the types without artworks.

Feel free to remove the sensitive data from the result since I just want to see the structure of API response.

**Comment by HueLiu** [comment]
```
[].concat.apply([], (await MusicKit.getInstance().api.music('v1/me/recommendations?timezone=%2B08:00&with=friendsMix%2Clibrary&name=listen-now&art%5Burl%5D=c%2Cf&omit%5Bresource%5D=autos&relate%5Beditorial-items%5D=contents&extend=editorialCard%2CeditorialVideo&extend%5Balbums%5D=artistUrl&extend%5Bplaylists%5D=artistNames%2CeditorialArtwork&extend%5Bsocial-profiles%5D=topGenreNames&include%5Balbums%5D=artists&include%5Bsongs%5D=artists&include%5Bmusic-videos%5D=artists&fields%5Balbums%5D=artistName%2CartistUrl%2Cartwork%2CcontentRating%2CeditorialArtwork%2CeditorialVideo%2Cname%2CplayParams%2CreleaseDate%2Curl&fields%5Bartists%5D=name%2Curl&extend%5Bstations%5D=airDate%2CsupportsAirTimeUpdates&meta%5Bstations%5D=inflectionPoints&types=artists%2Calbums%2Ceditorial-items%2Clibrary-albums%2Clibrary-playlists%2Cmusic-movies%2Cmusic-videos%2Cplaylists%2Cstations%2Cuploaded-audios%2Cuploaded-videos%2Cactivities%2Capple-curators%2Ccurators%2Ctv-shows&relate=catalog&platform=web')).data.data.map(recommendation => recommendation?.relationships?.contents?.data?.filter(item => !item?.attributes?.artwork)))
(2) [{…}, {…}]
0: {id: 'p.V7VY6QAuZRV7vWq', type: 'library-playlists', href: '/v1/me/library/playlists/p.V7VY6QAuZRV7vWq', attributes: {…}, relationships: {…}}
1: {id: 'p.ZOAXZl5h48GzbEO', type: 'library-playlists', href: '/v1/me/library/playlists/p.ZOAXZl5h48GzbEO', attributes: {…}, relationships: {…}}
length: 2
```

![image](https://user-images.githubusercontent.com/19755727/134492924-6aebec39-feea-45d9-a0f6-4d938a0a6953.png)

**Comment by lujjjh** [comment]
Thank you very much.

I think I find the problem. [LibraryPlaylists.Attributes](https://developer.apple.com/documentation/applemusicapi/libraryplaylists/attributes)' artwork is optional.

Believe #3 fixes this issue.

**Comment by lujjjh** [comment]
I've published [v0.1.0.0-beta.4](https://github.com/lujjjh/lito/releases/tag/v0.1.0.0-beta.4).

I'm hoping it has fixed this issue.

**Comment by HueLiu** [comment]
thx, now lito can play music normally.
